kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -102,59 +103,47 @@ define([
          * Returns a Promise that resolves once the rendering is done.
          */
         render: function () {
-            return this.updateView();
+            // a full render should wipe the entire view.
+            // really, this should only be done once. Subsequent data fetching
+            // just updates the data in the datatable.
+            this.$elem.empty();
+            this.renderFileHeader();
+            return this.updateView().then(() => {
+                this.renderImportButton();
+            });
         },
 
         updateView: function () {
-            return Promise.resolve(
-                this.stagingServiceClient.list({
-                    path: this.subpath,
-                })
-            )
+            return Promise.resolve(this.stagingServiceClient.list())
                 .then((data) => {
-                    //list is recursive, so it'd show all files in all subdirectories. This filters 'em out.
-                    const files = JSON.parse(data).filter((f) => {
-                        // this is less complicated than you think. The path is the username,
-                        // subpath, and name concatenated. The subpath may be empty so we
-                        // filter it out and only join defined things. If that's the same as
-                        // the file's path, we're at the right level. If not, we're not.
-                        return (
-                            [this.userInfo.user, this.subpath, f.name]
-                                .filter((p) => p.length > 0)
-                                .join('/') === f.path
-                        );
-                    });
-                    files.forEach((f) => {
-                        if (!f.isFolder) {
-                            f.imported = {};
-                        }
+                    const files = JSON.parse(data).map((f) => {
+                        f.path = f.path.substring(f.path.indexOf('/') + 1);
+                        f.subdir = '/' + f.path.substring(0, f.path.lastIndexOf('/')) || '/';
+                        return f;
                     });
                     return this.identifyImporterMappings(files);
                 })
                 .then((files) => {
                     const scrollTop = this.$elem.parent().scrollTop();
                     $(`.${fileMetadataCssBaseClass}`).detach();
-                    this.$elem.empty();
-                    this.renderFileHeader();
-                    this.renderFiles(files);
+
+                    if (this.fullDataTable) {
+                        this.fullDataTable.clear().rows.add(files).draw(false);
```

Comment:
From the DataTables docs: this wipes the table and adds all new data (which might be entirely different since the last refresh). `draw(false)` retains the table state - page, sorting, etc.